### PR TITLE
SGA-128-Friend-invites

### PR DESCRIPTION
**Pull request by GillWell**
## Summary of the Pull Request

SGA 128
Create a friend invite screen to show received friend requests, functions for accepting, declining, sending friend requests.

SGA 129
Create a screen to show current friends, functions to get the friends data and remove them

## PR Checklist

SGA 128
- [x] “Friends” screen is accessed from avatar click dropdown
- [x] Friend invite functions should be displayed inside a tab “Friend invites” within “Friends” screen
- [x] Friend invites tab should also display received invites list
- [x] Received invites can be accepted or denied
- [ ] (optional) User should be notified when he receives a friend invite (either through a system notification or a bubble near the icon where friends screen is located (see example below))

SGA 129
- [x] Friends list should be displayed inside a tab “Friends list” within “Friends” screen

## Detailed Description of the Pull Request / Additional comments
To send a friend request the user needs to use the search bar to first find the user then send a friend request. User who have sent a friend request will be showed below, the user can accept or decline. Can't send friend request to yourself, friends, users that have sent a friend request already. Friend list shows users that you are friends with. Able to remove friends

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -179,4 +179,21 @@
...
+    <string name="friend_request_sent">Friend request sent</string>
+    <string name="friend_removed">Friend removed</string>
+    <string name="friends_list">FRIENDS LIST</string>
+    <string name="friend_requests">FRIEND REQUESTS</string>
+    <string name="people_you_are_friends_with">People you are friends with</string>
+    <string name="you_don_t_have_any_friends_yet">You don\'t have any friends yet</string>
+    <string name="received_friend_requests">Received friend requests</string>
+    <string name="other_users_who_have_sent_you_friend_requests">Other users who have sent you friend requests</string>
+    <string name="search_for_a_user_to_display_them_here">Search for a user to display them here.</string>
+    <string name="Searched_user">User found by searching for their email address</string>
```
Comment (mantasjasikenas) [inline app/src/main/res/values/strings.xml]: Should start with lowercase

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -179,4 +179,21 @@
...
+    <string name="Searched_users">Searched user</string>
```
Comment (mantasjasikenas) [inline app/src/main/res/values/strings.xml]: Should start with lowercase

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -179,4 +179,21 @@
...
+    <string name="user_not_found">User not found</string>
+    <string name="Already_friends">You are already friends with this user</string>
```
Comment (mantasjasikenas) [inline app/src/main/res/values/strings.xml]: Should start with lowercase

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -179,4 +179,21 @@
...
+    <string name="Self_friend">Can\'t send a friend request to yourself!</string>
```
Comment (mantasjasikenas) [inline app/src/main/res/values/strings.xml]: Should start with lowercase

File: app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                    isButtonEnabled = !emails.contains(it.lowercase()) && it != userEmail
+                    nameError = if (it == userEmail) {
+                        context.getString(R.string.Self_friend)
+                    } else if (emails.contains(it.lowercase())) {
+                        context.getString(R.string.Already_friends)
+                    } else {
+                        null
+                    }
+                },
+                label = { Text("Send a friend request to") },
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt]: Move text to string resources

File: app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendsScreen.kt [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            selectedTabIndex = selectedTabIndex,
+        )
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(scrollState)
+                .padding(vertical = 16.dp)
+                .padding(bottom = 6.dp),
+            verticalArrangement = Arrangement.Center,
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendsScreen.kt]: Arrange elements in top of container (verticalArrangement = Arrangement.Top)

File: app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+            FriendRequestContent(
+                viewModel = viewModel,
+                uiState = uiState as FriendsUiState.Success,
+            )
+        }
+    }
+}
+
+@Composable
+fun FriendRequestContent(
```
Comment (mantasjasikenas) [inline app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendRequestScreen.kt:59]: Arrange elements in top of container

**Review by mantasjasikenas — CHANGES_REQUESTED** [review]

**Review by mantasjasikenas — APPROVED** [review]

**Comment by mantasjasikenas** [comment]
Please add missing strings in the Lithuanian language

File: app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendCard.kt [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+                Button(
+                    modifier = Modifier
+                        .padding(start = 6.dp),
+                    colors = ButtonDefaults.buttonColors(MaterialTheme.colorScheme.error),
+                    onClick = {
+                        coroutineScope.launch {
+                            friendsViewModel.removeFriend(user.uid)
+                        }
+                        SnackbarManager.showMessage(context.getString(R.string.friend_removed))
+                    },
```
Comment (paulius-mongirdas) [inline app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendCard.kt]: Maybe add confirmation dialog (like in market purchases) if user really wants to remove friend.

File: app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendsListScreen.kt
Comment (paulius-mongirdas) [inline app/src/main/java/com/github/k409/fitflow/ui/screen/friends/FriendsListScreen.kt:1]: Friends list is not refreshed when friend request is accepted. This seems to be an issue related to device API level, as it works fine on API 34, but does not on 29.

File: app/src/main/res/values-lt/strings.xml [hunk 9]
```diff
@@ -289,4 +289,5 @@
     <string name="info_app_usage">Norint užtikrinti, kad jūsų tikslai būtų pasiekti ir jūsų sveikata būtų optimizuota, būtina <b>susieti Health Connect</b> su kitomis programėlėmis, tokiomis kaip Google Fit ar Samsung Health. Be sinchronizacijos, jūsų tikslai gali būti nepasiekti, o bendra jūsų sveikata gali būti pavojinga. \n\nĮsitikinkite, kad <b>įdiegėte Health Connect</b> ir <b>susiejote ją</b> su jūsų mėgstamomis sveikatos programėlėmis, kad galėtumėte be sutrikimų stebėti savo tikslų įgyvendinimą.</string>
     <string name="health_connect">Health Connect</string>
     <string name="info_health_connect"><b>Health Connect</b> yra programėlė, skirta centralizuoti ir racionalizuoti sveikatos duomenis iš įvairių šaltinių, leidžianti vartotojams efektyviau stebėti ir valdyti savo sveikatos rodiklius. Ji veikia kaip mazgas, integruojantis duomenis iš skirtingų sveikatos ir fitneso programėlių, suteikiantis vartotojams išsamų savo sveikatos ir fitneso kelionės vaizdą.</string>
+    <string name="sure_to_remove_friend">Ar jūs tikrai norite pašilinti draugą </string>
```
Comment (paulius-mongirdas) [inline app/src/main/res/values-lt/strings.xml]: Spelling

**Review by paulius-mongirdas — APPROVED** [review]

**Pull request merged**